doc/source/topologies.rst

```diff
@@ -0,0 +1,137 @@
+Topologies
+==========
+
+
+Starting off
+-----------------
+Topology files are located in ``topologies`` in your streamparse project folder.
+There can be any number of topology files for your project in this directory.
+
+* topologies/my-topology.clj
+* topologies/my-other-topology.clj
+* topologies/my-third-topology.clj
+
+So on and so forth.
+
+Your topology file that describes the Directed Acyclic Graph (DAG) that will 
+be run in Storm.
+
+Looking at ``my-topology.clj``, we start off with importing the streamparse
+Clojure DSL functions
+
+.. code-block:: clojure
+
+    (ns my-topology
+      (:use     [streamparse.specs])
+      (:gen-class))
+
+Notice the ``my-topology`` matches the name of the file. We then have the
+import function.
+
+You could optionally avoid all of the streamparse-provided helper functions and 
+import your own functions or the Clojure DSL for Storm directly. 
+
+.. code-block:: clojure
+
+    (ns my-topology
+      (:use     [backtype.storm.clojure])
+      (:gen-class))
+
+In the next part of the file, we setup a topology definition, also named
+``my-topology`` (matching the ``ns`` line and filename). This definition is 
+actually a Clojure function that takes the topology options as a single map 
+argument. This function returns a list of 2 maps -- a spout map, and a bolt map.
+These two maps define the DAG that is your topology.
+
+.. code-block:: clojure
```

Comment:
Perhaps we should give the user a little help with the absolute basics of Clojure data definition. So, we could say.

Here's a quick Clojure overview so you don't get lost. `(defn fn-name [options] expressions)` defines a function called `fn-name` that takes `options` as an argument and evaluates each of the `expressions`, treating the last evaluated expression as a return value for the function. When defining a topology using one of these Clojure functions, you always return a Clojure list of 2 maps, as described above. A list in Clojure is specified as `[val1 val2 ... valN]`. A map is defined as `{"key-1" val1 "key-2" val2 ... "key-N" valN}`. For Python programmers, it can be a little tricky that whitespace is not significant and `,` is actually treated as whitespace, so that `[val1 val2]` and `[val1, val2]` are actually identical lists.

Last bits: `(python-spout-spec ...)` and `(python-bolt-spec ...)` are just convenience functions provided by streamparse for creating Topology components. and `;; comment` is just a line comment.